scripts/flow_chart.py

```diff
@@ -0,0 +1,138 @@
+        class_id = variant["recipe_quality"]
+        
+        if variant['machine'] == 'recycler':
+            class_id += '-recycling'
+            
+        return graph_id, class_id
+
+    def render_variant(self, recipe, variant_list, classes):
+        """Render a single recipe variant as a Mermaid subgraph."""
+        recipe_name = self.get_localized_name(recipe, self.recipes)
+        output_lines = []
+        
+        for variant in variant_list:
+            # Get localized names
+            machine = self.get_localized_name(variant['machine'], self.items)
+            name = self.get_localized_name(variant['recipe_name'], self.recipes)
+            
+            # Format modules string
+            mods = self.format_modules(variant['num_qual_modules'], 
+                                     variant['num_prod_modules'])
+            
+            # Generate IDs and track classes
+            graph_id, class_id = self.get_node_ids(recipe, variant)
+            classes[class_id].append(graph_id)
+            
+            # Create node definition
+            node_text = (f'{graph_id}[{name.replace("(","").replace(")", "")} - '
+                        f'{variant["recipe_quality"]} - {machine}{mods} x '
+                        f'{math.ceil(variant["solution_value"])}]')
+            output_lines.append(node_text)
+            
+        return f"subgraph {recipe_name}\n{'\n'.join(output_lines)}\nend\n"
+
+    def generate_class_definitions(self):
+        """Generate Mermaid class definitions for different quality levels."""
+        return """
+    classDef normal fill:#BCBCBC
+    classDef uncommon fill:#77E66A
+    classDef rare fill:#4890F2
+    classDef epic fill:#AF24F0
+    classDef legendary fill:#EC9736
+    classDef normal-recycling fill:#BCBCBC,stroke:#f5495a,stroke-width:4px
+    classDef uncommon-recycling fill:#77E66A,stroke:#f5495a,stroke-width:4px
+    classDef rare-recycling fill:#4890F2,stroke:#f5495a,stroke-width:4px
+    classDef epic-recycling fill:#AF24F0,stroke:#f5495a,stroke-width:4px
+    classDef legendary-recycling fill:#EC9736,stroke:#f5495a,stroke-width:4px"""
+
+    def write_flow_chart(self, output_flow_chart):
+        """Generate and save a Mermaid flow chart of the recipe system."""
+        variants = self.collect_recipe_variants()
+        classes = defaultdict(list)
+        
+        # Generate subgraphs for each recipe
+        subgraphs = [
+            self.render_variant(recipe, variant_list, classes)
+            for recipe, variant_list in variants.items()
+        ]
+        
+        # Generate class assignments
+        class_assignments = [
+            f"class {','.join(graph_ids)} {class_id}"
+            for class_id, graph_ids in classes.items()
+        ]
+        
+        # Combine all components into final sequence
+        sequence = (
+            "graph LR\n"
+            f"{'\n'.join(subgraphs)}"
```

Comment:
This line, and the one two lines below it, also had the same error `SyntaxError: f-string expression part cannot include a backslash`.